cli/src/main/java/com/liferay/blade/cli/util/ReleaseUtil.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.blade.cli.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.File;
+
+import java.time.temporal.ChronoUnit;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * @author Drew Brokke
+ */
+public class ReleaseUtil {
+
+	public static ReleaseEntry getReleaseEntry(String releaseKey) {
+		return _releaseUtil._releaseEntryMap.getOrDefault(releaseKey, _EMPTY_RELEASE_ENTRY);
+	}
+
+	public static void refreshReleases() {
+		System.out.println("Checking for new releases...");
+
+		_releaseUtil = new ReleaseUtil(0);
+	}
+
+	public static Stream<ReleaseEntry> releaseEntriesStream() {
+		return _releaseUtil._releaseEntries.stream();
+	}
+
+	public static <T> T withReleaseEntriesStream(Function<Stream<ReleaseEntry>, T> function) {
+		return function.apply(releaseEntriesStream());
+	}
+
+	public static <T> T withReleaseEntry(String releaseKey, Function<ReleaseEntry, T> function) {
+		return function.apply(getReleaseEntry(releaseKey));
+	}
+
+	public static class ReleaseEntry {
+
+		public String getProduct() {
+			return _product;
+		}
+
+		public String getProductGroupVersion() {
+			return _productGroupVersion;
+		}
+
+		public String getProductVersion() {
+			return _productVersion;
+		}
+
+		public String getReleaseKey() {
+			return _releaseKey;
+		}
+
+		public String getTargetPlatformVersion() {
+			return _targetPlatformVersion;
+		}
+
+		public String getUrl() {
+			return _url;
+		}
+
+		public boolean isPromoted() {
+			return _promoted;
+		}
+
+		@JsonProperty("product")
+		private String _product;
+
+		@JsonProperty("productGroupVersion")
+		private String _productGroupVersion;
+
+		@JsonProperty("productVersion")
+		private String _productVersion;
+
+		@JsonProperty("promoted")
+		private boolean _promoted;
+
+		@JsonProperty("releaseKey")
+		private String _releaseKey;
+
+		@JsonProperty("targetPlatformVersion")
+		private String _targetPlatformVersion;
+
+		@JsonProperty("url")
+		private String _url;
+
+	}
+
+	private ReleaseUtil(int maxAge) {
+		File releasesJsonFile = new File(_workspaceCacheDir, "releases.json");
+
+		_releaseEntries = ResourceUtil.readJson(
```

Comment:
I don't understand the ordering here?
1. ResourceUtil.getLocalFileResolver(releasesJsonFile, maxAge, ChronoUnit.DAYS),
2. ResourceUtil.getURLResolver(_workspaceCacheDir, "https://releases.liferay.com/releases.json", "releases.json"),
3. ResourceUtil.getURLResolver(_workspaceCacheDir, "https://releases-cdn.liferay.com/releases.json", releases.json"),
4. ResourceUtil.getLocalFileResolver(releasesJsonFile),
5. ResourceUtil.getClassLoaderResolver("/releases.json")

Would checking CDN not come before non-cdn?

Reply:
Makes sense - I copied from the order in workspace, which was preserved from when it read `.product_info.json`, but that may have been a mistake then too.

Reply:
So does this also mean that 

if local cache is too older (case #1) and both CDN and releases.liferay.com (case 2 and 3) fail to download (network error lets say), then is case#4 that it fails back to using the "too old for use" from case 1?  Should maybe then it promote case#5 (the built-in copy) over a "too old cached copy" as fallback?

And if we are trying to "test things out ahead of time", maybe there needs to be CASE-0 that looks at things from ENV var?

Reply:
> Should maybe then it promote case#5 (the built-in copy) over a "too old cached copy" as fallback?

Well, I figured that the local copy will almost always be newer than the built-in copy, but this may be an incorrect assumption.

> And if we are trying to "test things out ahead of time", maybe there needs to be CASE-0 that looks at things from ENV var?

This could be very useful! I'll play around with this.

Reply:
Ok I applied the re-ordering and added a check for the environment variable.